cmds/core/netcat/connect_test.go

```diff
@@ -20,6 +22,113 @@ import (
 	"github.com/u-root/u-root/pkg/netcat"
 )
 
+func TestConnectMode(t *testing.T) {
+	tests := []struct {
+		name        string
+		network     string
+		address     string
+		connectFunc func(output io.Writer, network, address string) error
+		expectError bool
```

Comment:
If we have test cases that throw known errors, wouldn't it make more sense to embed them into the test cases? So either it is `nil` or the error we predefined?

 ```suggestion
	     expectedError error
```

and then we can check for the result
```go
if err := cmd.connectMode(io.Discard, tt.network, tt.address); err != tt.expectedError {
    t.Errorf("wanted %v, got %v", tt.expectedError, err)
}
```

Reply:
If there are defined error values, yes, sure. This would make sense in the actual [connect func](https://github.com/u-root/u-root/pull/3291/files/2c13f0513578ad8168ab7f2c0a4ba427f78c643f#diff-1445756be1283b81274f42febc4561c59c56b8d236f6b49317e23eb5d394c3edR46) which is the former 'ConnectMode()' function. 

What I did here is pushing the code of the former 'ConnectMode()' function into 'connect()' and added logic in the new 'ConnectMode()' function to call 'connect()' several times in some cases. To test this logic I'm using a stub function instead of 'connect()' that just return errors as needed. I think this is sufficient to test the introduced logic. 

Adding defined error types and testing against them would be in the scope of [TestConnect](https://github.com/u-root/u-root/pull/3291/files/2c13f0513578ad8168ab7f2c0a4ba427f78c643f#diff-16aa8d474e6f1d06d5edc08b989c613ff40f2854257292332a816e3f325a8a9eR152) and is out of scope for this PR.

Reply:
I understand. Let's keep it the way it is.